diff --git a/CMakeLists.txt b/CMakeLists.txt
index 437570b..dd4d720 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,11 @@ if(MSVC)
 	cmake_policy(SET CMP0091 NEW)
 endif()
 
+execute_process(
+    COMMAND "./autogen.sh"
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
 project(LibreSSL LANGUAGES C ASM)
 
 include(CheckFunctionExists)
@@ -234,12 +239,6 @@ if(HAVE_REALLOCARRAY)
 	add_definitions(-DHAVE_REALLOCARRAY)
 endif()
 
-# XXX strcasecmp() is in strings.h which isn't available everywhere
-check_function_exists(strcasecmp HAVE_STRCASECMP)
-if(HAVE_STRCASECMP)
-	add_definitions(-DHAVE_STRCASECMP)
-endif()
-
 # Emscripten's strlcat and strlcpy triggers ASAN errors
 if(NOT EMSCRIPTEN)
 	check_symbol_exists(strlcat "string.h" HAVE_STRLCAT)
@@ -253,21 +252,6 @@ if(NOT EMSCRIPTEN)
 	endif()
 endif()
 
-check_symbol_exists(strndup "string.h" HAVE_STRNDUP)
-if(HAVE_STRNDUP)
-	add_definitions(-DHAVE_STRNDUP)
-endif()
-
-if(WIN32)
-	set(HAVE_STRNLEN true)
-	add_definitions(-DHAVE_STRNLEN)
-else()
-	check_symbol_exists(strnlen "string.h" HAVE_STRNLEN)
-	if(HAVE_STRNLEN)
-		add_definitions(-DHAVE_STRNLEN)
-	endif()
-endif()
-
 check_symbol_exists(strsep "string.h" HAVE_STRSEP)
 if(HAVE_STRSEP)
 	add_definitions(-DHAVE_STRSEP)
