diff --git a/src/raylib.h b/src/raylib.h
index e5846985..6c890d4f 100644
--- a/src/raylib.h
+++ b/src/raylib.h
@@ -1330,7 +1330,7 @@ RLAPI Rectangle GetCollisionRec(Rectangle rec1, Rectangle rec2);
 RLAPI Image LoadImage(const char *fileName);                                                             // Load image from file into CPU memory (RAM)
 RLAPI Image LoadImageRaw(const char *fileName, int width, int height, int format, int headerSize);       // Load image from RAW file data
 RLAPI Image LoadImageAnim(const char *fileName, int *frames);                                            // Load image sequence from file (frames appended to image.data)
-RLAPI Image LoadImageAnimFromMemory(const char *fileType, const unsigned char *fileData, int dataSize, int *frames); // Load image sequence from memory buffer
+RLAPI Image LoadImageAnimFromMemory(const char *fileType, const unsigned char *fileData, int dataSize, int *frames, int **delays); // Load image sequence from memory buffer
 RLAPI Image LoadImageFromMemory(const char *fileType, const unsigned char *fileData, int dataSize);      // Load image from memory buffer, fileType refers to extension: i.e. '.png'
 RLAPI Image LoadImageFromTexture(Texture2D texture);                                                     // Load image from GPU texture data
 RLAPI Image LoadImageFromScreen(void);                                                                   // Load image from screen buffer and (screenshot)
diff --git a/src/rtextures.c b/src/rtextures.c
index 81d615c3..0e3ceba6 100644
--- a/src/rtextures.c
+++ b/src/rtextures.c
@@ -379,8 +379,8 @@ Image LoadImageAnim(const char *fileName, int *frames)
 //  - Image.data buffer includes all frames: [image#0][image#1][image#2][...]
 //  - Number of frames is returned through 'frames' parameter
 //  - All frames are returned in RGBA format
-//  - Frames delay data is discarded
-Image LoadImageAnimFromMemory(const char *fileType, const unsigned char *fileData, int dataSize, int *frames)
+//  - Frames delay data is NOT discarded
+Image LoadImageAnimFromMemory(const char *fileType, const unsigned char *fileData, int dataSize, int *frames, int **delays)
 {
     Image image = { 0 };
     int frameCount = 0;
@@ -394,13 +394,10 @@ Image LoadImageAnimFromMemory(const char *fileType, const unsigned char *fileDat
         if (fileData != NULL)
         {
             int comp = 0;
-            int *delays = NULL;
-            image.data = stbi_load_gif_from_memory(fileData, dataSize, &delays, &image.width, &image.height, &frameCount, &comp, 4);
+            image.data = stbi_load_gif_from_memory(fileData, dataSize, delays, &image.width, &image.height, &frameCount, &comp, 4);
 
             image.mipmaps = 1;
             image.format = PIXELFORMAT_UNCOMPRESSED_R8G8B8A8;
-
-            RL_FREE(delays);        // NOTE: Frames delays are discarded
         }
     }
 #else
